### DISCUSSION TITLE
Deploy a provider in a different namespace than the core

### DISCUSSION THREAD

**Discussion by xoanmi** [crossplane/crossplane#5449] [General]
Hi folks!

It is possible to deploy a provider in a different namespace than the core?

For security and management reasons, we would like to split the deployment of each provider into its namespace. Is this possible in any way?

**Comment by jbw976** [comment]
it looks like this should be supported with the `--namespace` flag that can be passed as an arg to the crossplane core pod.  From the crossplane binary help output:
```
  -n, --namespace="crossplane-system"     Namespace used to unpack and run packages ($POD_NAMESPACE).
```

However, I just tried that out and it doesn't seem to work.  I created a namespace called `crossplane-providers` and then installed the Crossplane helm chart with:
```
helm install crossplane \
--namespace crossplane-system \
--create-namespace crossplane-stable/crossplane \
--set args='{"--namespace=crossplane-providers"}'
```

Then installed a provider with:
```
❯ cat <<EOF | kubectl apply -f -
apiVersion: pkg.crossplane.io/v1
kind: Provider
metadata:
  name: provider-aws-s3
spec:
  package: xpkg.upbound.io/upbound/provider-aws-s3:v1.1.0
EOF
```

The package does not get to healthy, as the package manager appears to be failing to [look up the crossplane service account](https://github.com/crossplane/crossplane/blob/release-1.15/internal/controller/pkg/revision/reconciler.go#L930-L935), so that it can reuse any of its image pull secrets on the newly installed provider too:

```
❯ crossplane beta trace provider.pkg.crossplane.io/provider-aws-s3 -o wide
NAME                                               PACKAGE                                   VERSION   INSTALLED   HEALTHY   STATE    STATUS
Provider/provider-aws-s3                           xpkg.upbound.io/upbound/provider-aws-s3   v1.1.0    True        False     -        UnhealthyPackageRevision: cannot prepare runtime manifest builder options: cannot get Crossplane service account: ServiceAccount "crossplane" not found
└─ ProviderRevision/provider-aws-s3-13463c1ac7d2   xpkg.upbound.io/upbound/provider-aws-s3   v1.1.0    -           False     Active   UnhealthyPackageRevision: cannot prepare runtime manifest builder options: cannot get Crossplane service account: ServiceAccount "crossplane" not found
```

That seems like a bug where the package manager is looking in the wrong namespace for the crossplane service account - it's looking in the new namespace where providers will be installed into, not the namespace where it itself is running. This typically works fine because the default for both is the same 😞 

TL;DR it looks like this should work, but a potential bug is preventing it from working correctly

**Reply by xoanmi** [reply]
Should create a bug then?

**Reply by jbw976** [reply]
Created https://github.com/crossplane/crossplane/issues/5576 from this discussion, thanks again for bringing it up @xoanmi 🙇‍♂️
